system-design/youtube-seek-to.md

```diff
@@ -0,0 +1,29 @@
+# Youtube API - seekTo()
+
+## Overview
+`seekTo()` function is available on the player object and is used to play the video from the timestamp specified - `seekTo(timestamp)`.
+
+## Use Cases  
+seekTo() function can be used in the following scenarios  
+* When a user joins late 
+
+    When the video starts playing the video sync service on the client side will keep track of current play progress. The progress will be shared to the backend by all clients and the backend video sync service will check if all users are on the same progress. When a user joins late, on joining, the client video sync service will get the current video progress(timestamp) from the backend video sync service. On recieveing the timestamp, `seekTo()` will be executed post which `playVideo()` will be executed.This will sync the late joining user to the current video progress of the group.  
+
+* When user changes the progress of the video  
+
+    When the host of the group decides to play video from a non-zero timestamp, i.e the user changes progress bar position. An event* is fired that communicates the current progress to the backend which further communicates this with all the other connected clients. Other connected clients on recieving this information will execute `seekTo()` followed by `playVideo()`.
+    
+### Flow Diagram
+![alt text](../flow-diagrams/seekTo-flow.png)
+[Lucid chart link](https://lucid.app/lucidchart/invitations/accept/inv_2a7ff478-576c-4502-951f-c550e590a4c8?viewport_loc=-897%2C-173%2C3640%2C1785%2C0_0)
```

Comment:
Following format would be a better fit for this diagram.
https://www.geeksforgeeks.org/unified-modeling-language-uml-activity-diagrams/

Reply:
@chirgjn you want me to redo the diagram as an activity diagram?

Reply:
I am not looking for 100% conformance to the guidelines of activity diagram.
I think we can do better in terms of being explicit about the order of operations in the current diagram.

This might also be a useful read: https://www.geeksforgeeks.org/unified-modeling-language-uml-sequence-diagrams

One more suggestion: You can have multiple diagrams. One for each flow or One to describe order and one to describe data flow.